TestColorMap.cpp

```diff
@@ -32,10 +32,10 @@ TEST(ColorMapTest, TestSpecificColorPairFormatting) {
 
     // Check specific color pair formatting
     std::ostringstream colorPairStream;
-    for (size_t i = 1; i < colorPairs.size(); ++i) {
-        colorPairStream << i << " | "
-            << colorPairs[i].second << " | "
-            << colorPairs[i].first << '\n';
+    for (size_t i = 1; i <= colorPairs.size(); ++i) {

```

Comment:
Instead of re-implementing the output in the test, an alternate way is to see what a customer wants in the output:
- clear header
- formatted string of one row must have two commas
- content of each row must be correct
- total number of rows must be 25

What's the advantage of this alternative? Testing this way can guide us to do 'single responsibility' - by forcing every test to be independent, we are also forcing the implementation to split accordingly: function to format one row will be separate from the function that makes the headings, etc.